04 - Guides, Workflows, & Courses/Guides/How to add automated tests to your plugin.md

```diff
@@ -1,21 +1,422 @@
+
+This guide will use the [Obsidian example plugin](https://github.com/obsidianmd/obsidian-sample-plugin) as a showcase. It also assumes knowledge about unit testing to a certain degree. If you wish to go through the guide step-by-step, please download and enable the example plugin like described in [Build a plugin](https://docs.Obsidian.md/Plugins/Getting+started/Build+a+plugin). 
+
+This guide focused on how to integrate Jest to your plugin and what steps to take to deal with the Obsidian API when running plugin code in a test environment. It does not provide guidance how to write test cases or what good tests are. Please refer to [[Plugin Testing for Developers]] for a deeper look at these topics as well as [[How to find examples of Jest-based plugin tests]] for examples.
+
+## Integrate Jest into Obsidian-example-plugin
+
+Integrating Jest to a project and make it compatible with typescript code requires some additional dependencies.
+
+1. Open up a terminal and navigate to your plugins root folder
+2. Install Jest: `npm install --save-dev jest`
+3. Install types for correct typing in typescript files: `npm install --save-dev @types/jest`
+4. Install ts-jest to transpile typescript files: `npm install --save-dev ts-jest`
+5. Create a Jest configuration file by running `npx ts-jest config:init`
+6. Open `tsconfig.json` and add `"esModuleInterop": true` as an additional property to `compilerOptions` 
+7. Open `package.json` and add `"test": "jest"` to the `scripts` section 
+
+Jest is now integrated into the project and usable for testing.
+
+## Executing a first automated test
+
+> [!hint] Structuring with folders
+> For reason of simplicity, this guide adds files to the root folder. Generally, it is a good idea to structure your code in folders, i.e. productive code in `src` and tests in `tests` to keep your code base better maintainable. 
+
+1. Create a new file named `example.spec.ts` to your projects root folder
+
+> [!info] Test file names
+> It is not required to call tests with a `.spec.ts` ending, `.ts` is enough. However, it allows you to see at one glance which file contain productive code and which are test files, which will help with development and maintainability. It is also common to end test file names with `.test.ts` instead of `.spec.ts`.
+
+2. In `example.spec.ts`, add a `describe` block. Read more about describe blocks in [Jests documentation](https://jestjs.io/docs/api#describename-fn). Basically, a `describe` bundles up multiple test cases for better readability. 
+
+```ts
+// example.spec.ts
+describe('MyPlugin Tests', () => {
+
+});
+```
+
+3. To the describe, add a test case. Please note that `it` and `test` are aliases, so you can also call `test`, depending on your preference. Read more in [Jests documentation](https://jestjs.io/docs/api#testname-fn-timeout)
+
+```ts
+// example.spec.ts
+describe('MyPlugin Tests', () => {
+
+  it('should be able to execute test', () => {
```

Comment:
I will go ahead and merge, but looking at this file in PyCharm, these are among several that have `NBSP` characters rather than ordinary spaces...

It probably won't matter in the published docs though.